providers/desec/desecProvider.go

```diff
@@ -0,0 +1,223 @@
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Cannot(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUsePTR:              providers.Unimplemented(),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanAutoDNSSEC:          providers.Cannot(),
+}
+
+var defaultNameServerNames = []string{
+	"ns1.desec.io",
+	"ns2.desec.org",
+}
+
+func init() {
+	providers.RegisterDomainServiceProviderType("DESEC", NewDeSec, features)
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (c *api) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	return models.ToNameservers(defaultNameServerNames)
+}
+
+func (c *api) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	existing, err := c.GetZoneRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+	models.PostProcessRecords(existing)
+	clean := PrepFoundRecords(existing)
+	PrepDesiredRecords(dc)
+	return c.GenerateDomainCorrections(dc, clean)
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (c *api) GetZoneRecords(domain string) (models.Records, error) {
+	records, err := c.getRecords(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert them to DNScontrol's native format:
+	existingRecords := []*models.RecordConfig{}
+	for _, rr := range records {
+		existingRecords = append(existingRecords, nativeToRecords(rr, domain)...)
+	}
+	return existingRecords, nil
+}
+
+// EnsureDomainExists returns an error if domain doesn't exist.
+func (c *api) EnsureDomainExists(domain string) error {
+	if err := c.fetchDomainList(); err != nil {
+		return err
+	}
+	// domain already exists
+	if _, ok := c.domainIndex[domain]; ok {
+		return nil
+	}
+	return c.createDomain(domain)
+}
+
+// PrepFoundRecords munges any records to make them compatible with
+// this provider. Usually this is a no-op.
+func PrepFoundRecords(recs models.Records) models.Records {
+	// If there are records that need to be modified, removed, etc. we
+	// do it here.  Usually this is a no-op.
+	return recs
+}
+
+// PrepDesiredRecords munges any records to best suit this provider.
+func PrepDesiredRecords(dc *models.DomainConfig) {
+	// Sort through the dc.Records, eliminate any that can't be
+	// supported; modify any that need adjustments to work with the
+	// provider.  We try to do minimal changes otherwise it gets
+	// confusing.
+
+	dc.Punycode()
+	recordsToKeep := make([]*models.RecordConfig, 0, len(dc.Records))
+	for _, rec := range dc.Records {
+		if rec.Type == "ALIAS" {
+			// deSEC does not permit ALIAS records, just ignore it
+			printer.Warnf("deSEC does not support alias records\n")
+			continue
+		}
+		if rec.TTL < 3600 {
+			if rec.Type != "NS" {
+				printer.Warnf("deSEC does not support ttls < 3600. Setting ttl of %s type %s from %d to 3600\n", rec.GetLabelFQDN(), rec.Type, rec.TTL)
```

Comment:
deSEC supports lower TTLs if you ask nicely. Maybe there can be a hint in this message to contact support@desec.io?

Reply:
I could give a hint in the message, but for dnscontrol it would not make sense.
Maybe i could implement the validation of the ttl based on the response minimum_ttl, this would allow users to utilize this exception.

Could you allow lower ttls for the zone d3luxee.de so i can test this feature?

Reply:
Maybe rephrasing the message to "your deSEC account does not support" can give the user a better hint of what's going on?

I did update the minimum ttl for d3luxee.de.

Reply:
I did implement this check based on the minimum_ttl yesterday and adjusted the message.
The current "uncommited" message looks like this:
`WARNING: Please contact support@desec.io if you need ttls < 3600. Setting ttl of autodiscover.d3luxee.de type A from 60 to 3600`

Do you have any suggestion for the message?

Reply:
lgtm, thanks!

Reply:
I've tested and pushed the latest changes, dnscontrol will now validate and correct against the minimum_ttl of the specific domain instead of the hardcoded 3600 seconds.

This handles exceptions made by you or changes on the default minimum_ttl.